extensions/shared/src/main/java/app/revanced/extension/youtube/patches/shorts/ShortsPatch.java

```diff
@@ -221,4 +234,87 @@ private static void setNavigationBarLayoutParams(@NonNull ShortsPlayerState shor
         );
     }
 
+    /**
+     * Injection point for the "More" button in the Shorts player
+     */
+    public static void showShortsToolbarMenu(String enumString, View toolbarView) {
+        if (!Settings.HOOK_MORE_BUTTON.get())
+            return;
+        if (!isMoreButton(enumString))
+            return;
+        ImageView imageView = getChildView((ViewGroup) toolbarView, view -> view instanceof ImageView);
+        if (imageView == null)
+            return;
+
+        imageView.setOnLongClickListener(button -> {
+            showMoreButtonDialog(toolbarView.getContext());
+            return true;
+        });
+    }
+
+    private static void showMoreButtonDialog(Context context) {
+        AlertDialog.Builder builder = new AlertDialog.Builder(context);
+        builder.setTitle(getString("revanced_hook_more_button_dialog_title"));
+
+        String copyLink = getString("revanced_hook_more_button_option_copy_link_title");
+        String copyLinkTimestamp = getString("revanced_hook_more_button_option_copy_link_timestamp_title");
+        String downloadShort = getString("revanced_hook_more_button_option_download_short_title");
+        String openInNormalPlayer = getString("revanced_hook_more_button_option_open_in_normal_player_title");
+
+        String videoId;
+        if (ShortsInformation.getShortId() != null) {
+            // not in incognito mode
+            videoId = ShortsInformation.getShortId();
+        } else {
+            // in incognito mode, use videoId and hope it's correct
+            videoId = VideoInformation.getVideoId();
+        }
+
+        List<String> optionsList = new ArrayList<>();
+        Map<String, Runnable> actions = new HashMap<>();
+
+        if (Settings.COPY_URL_SHORT_TOOLBAR_MENU.get()) {
+            optionsList.add(copyLink);
+            actions.put(copyLink, () -> VideoUtils.copyUrl(false, videoId));
+        }
+
+        if (Settings.COPY_URL_WITH_TIMESTAMP_SHORT_TOOLBAR_MENU.get()) {
+            optionsList.add(copyLinkTimestamp);
+            actions.put(copyLinkTimestamp, () -> VideoUtils.copyUrl(true, videoId));
+        }
+
+        if (Settings.DOWNLOAD_SHORT_TOOLBAR_MENU.get()) {
+            optionsList.add(downloadShort);
+            actions.put(downloadShort, () -> VideoUtils.launchVideoExternalDownloader(videoId));
+        }
+
+        if (Settings.OPEN_IN_NORMAL_PLAYER_SHORT_TOOLBAR_MENU.get()) {
```

Comment:
This option shouldn't be available if `Open link directly` is not turned on

Reply:
will reflect this tomorrow